src/extensions/jana/JOmniFactory.h

```diff
@@ -34,7 +34,9 @@ class JOmniFactory : public JMultifactory {
 
     struct InputBase {
         std::string type_name;
-        std::string collection_name;
+        std::vector<std::string> collection_names;
+        bool is_variadic = false;
```

Comment:
I keep wondering if `is_variadic` can be `collection_names.size() > 1`...

Reply:
In principle, yes. At least until we decide we also need multi-collection inputs with fixed multiplicity (which one person has already asked for!)

Reply:
If it's multicollection with fixed multiplicity, it can just be repeated. Not sure I agree that's something that needs implementing. Or is there something I'm missing?

Reply:
I'm also leaning against it, just not rejecting it yet. 

The use case is as follows: Imagine a detector that has separate layers. Each layer has its own (subset) collection. The algorithm should be designed for an arbitrary number of layers, but when running EICrecon we know there will be exactly four. Right now we can either have four separate PodioInputs and put their collection pointers in a vector to pass to the algorithm, or we can have one VariadicPodioInput and pass it directly, at the cost of constraining what our other inputs can be. If we could state that there will be exactly four collections in the Input, there would be no constraints on the input and no manual repackaging step.

Reply:
Right now I favor sticking to three flavors of Input: Optional, ExactlyOne, and Variadic. They correspond nicely to the relations in UML: 0..1, 1, *

Reply:
Yes, and at some point we'll write the UML exporter for the whole set of factories :-)